clients/src/main/java/org/apache/kafka/clients/admin/internals/AdminMetadataManager.java

```diff
@@ -287,4 +293,12 @@ public void update(Cluster cluster, long now) {
             this.cluster = cluster;
         }
     }
+
+    /**
+     * Rebootstrap metadata with the cluster previously used for bootstrapping.
+     */
+    public void rebootstrap(long now) {
+        log.info("Rebootstrapping with {}", this.bootstrapCluster);
+        update(bootstrapCluster, now);
```

Comment:
Same question as with NetworkClient, don't we want to re-resolve bootstrap?

Reply:
It uses the same `ClientUtils.parseAndValidateAddresses` function inside, so the same logic applies.